assembly/std/arguments.ts

```diff
@@ -127,28 +181,57 @@ export class Args {
    */
   add<T>(arg: T): Args {
     if (arg instanceof Address) {
-      this.serialized = this.serialized.concat(arg.toStringSegment());
+      let length = new Uint8Array(1);
+      length[0] = arg._value.length;
```

Comment:
you should use `arg.toStringSegment()` instead of arg._value.

I think here you need `this.add<u32>(arg.toStringSegment());`. unless the length is auto converted to uint8.